tools/lib/runPipeline.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+var Cesium = require('cesium');
+var fsExtra = require('fs-extra');
+var path = require('path');
+var Promise = require('bluebird');
+var gzipTileset = require('./gzipTileset');
+
+var defaultValue = Cesium.defaultValue;
+var defined = Cesium.defined;
+var DeveloperError = Cesium.DeveloperError;
+
+var fsExtraCopy = Promise.promisify(fsExtra.copy);
+var fsExtraEmptyDir = Promise.promisify(fsExtra.emptyDir);
+var fsExtraRemove = Promise.promisify(fsExtra.remove);
+
+module.exports = runPipeline;
+
+/**
+ * Run an input pipeline.
+ *
+ * @param {Object} pipeline A JSON object containing an input, output, and list of stages to run.
+ * @param {String} pipeline.input Input tileset path.
+ * @param {String} [pipeline.output] Output tileset path.
+ * @param {Array<Object|String>} [pipeline.stages] The stages to run on the tileset.
+ * @param {Object} [options] An object with the following properties:
+ * @param {Boolean} [options.verbose] If true prints out debug messages to the console.
+ */
+function runPipeline(pipeline, options) {
+    pipeline = defaultValue(pipeline, defaultValue.EMPTY_OBJECT);
+    var inputPath = pipeline.input;
+    var outputPath = pipeline.output;
+    var stages = pipeline.stages;
+    if (!defined(inputPath)) {
+        throw new DeveloperError('pipeline.input is required');
+    }
+
+    inputPath = path.normalize(inputPath);
+    outputPath = path.normalize(defaultValue(outputPath,
+        path.join(path.dirname(inputPath), path.basename(inputPath) + '-processed')));
+
+    if (!defined(stages)) {
+        return fsExtraCopy(inputPath, outputPath);
+    }
+
+    options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+    var verbose = defaultValue(options.verbose, false);
+
+    var workingDirectory1 = path.join(path.dirname(inputPath), path.basename(inputPath) + '-working1');
```

Comment:
Here and below would be more robust if we use a working directory name that we know doesn't exist.